torchvision/transforms/v2/_transform.py

```diff
@@ -32,53 +33,29 @@ def _transform(self, inpt: Any, params: Dict[str, Any]) -> Any:
         raise NotImplementedError
 
     def forward(self, *inputs: Any) -> Any:
-        flat_inputs, spec = tree_flatten(inputs if len(inputs) > 1 else inputs[0])
+        sample = inputs if len(inputs) > 1 else inputs[0]
+        if is_simple_tensor(sample):
+            sample = datapoints.Image(sample)
+            simple_tensor_image_fallback = True
+        else:
+            simple_tensor_image_fallback = False
+
+        flat_inputs, spec = tree_flatten(sample)
 
         self._check_inputs(flat_inputs)
 
-        needs_transform_list = self._needs_transform_list(flat_inputs)
-        params = self._get_params(
-            [inpt for (inpt, needs_transform) in zip(flat_inputs, needs_transform_list) if needs_transform]
-        )
+        params = self._get_params(flat_inputs)
 
         flat_outputs = [
-            self._transform(inpt, params) if needs_transform else inpt
-            for (inpt, needs_transform) in zip(flat_inputs, needs_transform_list)
+            self._transform(inpt, params) if isinstance(inpt, self._transformed_types) else inpt for inpt in flat_inputs
```

Comment:
Since we no longer care about simple tensors here, we can use the regular `isinstance` and can potentially eliminate

https://github.com/pytorch/vision/blob/c73411a4dc785997015501625e3888699f6530aa/torchvision/prototype/transforms/utils.py#L52